docs/proposed/psa-driver-interface.md

```diff
@@ -0,0 +1,627 @@
+PSA Cryptoprocessor Driver Interface
+====================================
+
+This document describes an interface for cryptoprocessor drivers in the PSA cryptography API. This interface complements the PSA Cryptography API specification, which describes the interface between a PSA Cryptography implementation and an application.
+
+This specification is work in progress and should be considered to be in a beta stage. There is ongoing work to implement this interface in Mbed TLS, which is the reference implementation of the PSA Cryptography API. At this stage, Arm does not expect major changes, but minor changes are expected based on experience from the first implementation and on external feedback.
+
+Time-stamp: "2020/08/03 11:42:58 GMT"
+
+## Introduction
+
+### Purpose of the driver interface
+
+The PSA Cryptography API defines an interface that allows applications to perform cryptographic operations in a uniform way regardless of how the operations are performed. Under the hood, different keys may be processed in different hardware or in different logical partitions, and different algorithms may involve different hardware or software components.
```

Comment:
Why?

Reply:
When I read "different keys may be processed" I am thinking about key generation, computation rather than using a key for a cryptographic operation. Thus I proposed to add "may be used".

Reply:
Weird, for me it's rather the opposite: performing an operation that uses a key is “processes” since it takes the key as input, whereas for key generation it's borderline since the key isn't really “processed” but produced. “Used and processed” feels redundant to me. How about “stored and used”?

Reply:
That's the beauty of human language.
In fact, "processed" alone means a bit both to me thus with "processed and produced" I'd understand "processed" the way you do. Thus what about "processed and produced" then?